File: template/springboot/TinyPro/src/main/java/com/TinyPro/entity/po/Permission.java
package com.TinyPro.entity.po;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "permission")
public class Permission implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @TableId(type = IdType.AUTO)
    private Integer id;

    @NotBlank // 描述非空
    @Column(name = "`desc`", nullable = false)
    @TableField("`desc`")
    private String desc;

    @NotBlank // 权限名称非空
    private String name;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getDesc() {
        return desc;
    }

    public void setDesc(String desc) {
        this.desc = desc;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
}

File: template/nestJs/libs/models/src/permission.ts
import { ApiProperty } from '@nestjs/swagger';
import { Column, Entity, PrimaryGeneratedColumn } from 'typeorm';

@Entity('permission')
export class Permission {
  @ApiProperty({
    description: '权限ID'
  })
  @PrimaryGeneratedColumn()
  id: number;

  @ApiProperty({
    description: '权限简介'
  })
  @Column()
  desc: string;
  @ApiProperty({
    description: '权限名, 前端或外部服务使用'
  })
  @Column()
  name: string;
}
